.github/workflows/publish-pacts.yaml

```
@@ -0,0 +1,50 @@
+name: Publish Pacts
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ dev ]
+
+jobs:
+  setup-and-test:
+    runs-on: ubuntu-latest
+    outputs:
+      nb-b64: ${{ steps.encode.outputs.nb-b64 }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install yarn
+        run: |
+          yarn install
+
+      - name: Run tests
+        run: yarn test
+
+      - name: Encode the pact as non-breaking base 64 string
+        id: encode
+        env:
+          PACT_FULL_PATH: 'pacts/terra-ui-cbas.json'  # this is a temporary hack until WM-1858 is done.
```

Just confirming, right now this means only cbas will be able to do contract testing? How much overhead is it for other services to add?

---

not necessarily - it just means that a single workflow step is limited to one pact file. more contracts could be added with more workflow steps. the long term fix is to update the workflow `publish-contracts.yaml` (dispatched from `terra-github-workflows`) to accept a directory of pacts. There's a known solution for that - just have to get a hold of the right people to approve that change.

---

Nit: it's not obvious what the "temporary hack" is. Could this be elaborated on?

```suggestion
          PACT_FULL_PATH: 'pacts/terra-ui-cbas.json'  # Currently, workflows are limited to a single Pact file (see WM-1858).
```